test/gtest/uct/test_wakeup.cc

```diff
@@ -101,3 +108,224 @@ UCS_TEST_P(test_uct_wakeup, am)
+                           << completed_evt);
+        }
+    }
+
+    void cleanup() {
+        delete(m_buf1);
+        delete(m_buf2);
+        if (wakeup_handle) {
+            uct_wakeup_close(wakeup_handle);
+        }
+        test_uct_wakeup::cleanup();
+    }
+
+protected:
+    uct_wakeup_h wakeup_handle;
+    struct pollfd wakeup_fd;
+    size_t length;
+    uint64_t test_ib_hdr;
+    mapped_buffer *m_buf1, *m_buf2;
+    static size_t bcopy_pack_count;
+};
+
+size_t test_uct_wakeup_ib::bcopy_pack_count = 0;
+
+
+UCS_TEST_P(test_uct_wakeup_ib, tx_cq)
+{
+    ucs_status_t status;
+
+    initialize();
+
+    status = uct_wakeup_efd_arm(wakeup_handle);
+    ASSERT_EQ(status, UCS_OK);
+
+    /* check initial state of the fd and [send|recv]_cq */
+    EXPECT_EQ(poll(&wakeup_fd, 1, 0), 0);
+    check_send_cq(m_e1->iface(), 0);
+    check_recv_cq(m_e1->iface(), 0);
+
+    /* send the data */
+    send_msg_e1_e2();
+
+    /* make sure the file descriptor is signaled once */
+    ASSERT_EQ(poll(&wakeup_fd, 1, 1), 1);
+
+    status = uct_wakeup_efd_arm(wakeup_handle);
+    ASSERT_EQ(status, UCS_OK);
+
+    /* make sure [send|recv]_cq handled properly */
+    check_send_cq(m_e1->iface(), 1);
+    check_recv_cq(m_e1->iface(), 0);
+}
+
+
+UCS_TEST_P(test_uct_wakeup_ib, txrx_cq)
+{
+    ucs_status_t status;
+
+    initialize();
+
+    status = uct_wakeup_efd_arm(wakeup_handle);
+    ASSERT_EQ(status, UCS_OK);
+
+    /* check initial state of the fd and [send|recv]_cq */
+    EXPECT_EQ(poll(&wakeup_fd, 1, 0), 0);
+    check_send_cq(m_e1->iface(), 0);
+    check_recv_cq(m_e1->iface(), 0);
+
+    /* send the data */
+    send_msg_e1_e2();
+    send_msg_e2_e1();
+
+    twait(50); /* Let completion to be generated */
+
+    /* make sure the file descriptor is signaled */
+    ASSERT_EQ(poll(&wakeup_fd, 1, 1), 1);
+
+    status = uct_wakeup_wait(wakeup_handle);
+    ASSERT_EQ(status, UCS_OK);
+
+    /* make sure [send|recv]_cq handled properly */
+    check_send_cq(m_e1->iface(), 1);
+    check_recv_cq(m_e1->iface(), 1);
+
+}
+
+
+UCS_TEST_P(test_uct_wakeup_ib, multi_txrx_cq)
+{
+    const size_t msg_count = 5;
+    ucs_status_t status;
+
+    initialize();
+
+    status = uct_wakeup_efd_arm(wakeup_handle);
+    ASSERT_EQ(status, UCS_OK);
+
+    /* check initial state of the fd and [send|recv]_cq */
+    EXPECT_EQ(poll(&wakeup_fd, 1, 0), 0);
+    check_send_cq(m_e1->iface(), 0);
+    check_recv_cq(m_e1->iface(), 0);
+
+    /* send the data */
+    send_msg_e1_e2(msg_count);
+    send_msg_e2_e1(msg_count);
+
+    while ((test_uct_wakeup::am_handler_count    < msg_count) ||
+           (test_uct_wakeup_ib::bcopy_pack_count < msg_count)) {
+        progress();
+    }
+
+
+    /* make sure the file descriptor is signaled */
+    ASSERT_EQ(poll(&wakeup_fd, 1, 1), 1);
+
+    status = uct_wakeup_wait(wakeup_handle);
+    ASSERT_EQ(status, UCS_OK);
+
+    /* make sure the file descriptor is cleared */
+    ASSERT_EQ(poll(&wakeup_fd, 1, 1), 0);
```

Comment:
@brminich the `fd` won't be signaled after `uct_wakeup_efd_arm()` (as part of `uct_wakeup_wait()`) as you expected.